### PR TITLE
Only generate unique hash from allowed keys

### LINKED ISSUE #360: options.store with circular references causes exception
We are using Ember 3.7 at the moment and recently upgraded from 1.0.0 to 1.3.2. This behavior did not exist in 1.0.0 and was only introduced after the upgrade.

If you pass in a custom store into options: `this.model.infinity('model-name', { store: this.store })` you will get an exception `Maximum call stack size exceeded`. This is because the store has at least one property that has circular reference to itself, the `defaultAdapter` property. When generating a unique identifier using `stringifyObjectValues` you will essentially get into an infinite recursive call because of this circular reference. This is because the method recursively calls itself if the value of an option is `typeof` object (which the DS.Store is). So essentially you get to the store key which returns the store which has a key defaultAdapter that is also an object and has a key of store (and essentially you go store, defaultAdapter, store, defaultAdapter, store, defaultAdapter and on until call stack gets too big).

I know you don't need to pass this.store as it is the default I was only using it as an example. We are using HasManyQueryMixin which adds query method to has many relationships so that we can query relationships through parent model. So in our actual case we are passing the model as the store (I'm assuming Ember.Model also has some store of circular reference as one or more of it's properties).

Looks like this was introduced in https://github.com/ember-infinity/ember-infinity/commit/22bdd0578ffaa109189e061613d1fa8f3dbf0e06

Not sure what the best course of action is, clearly the purpose of the commit make sense (to include nested params in the unique identifier). Is the store relevant to generating a unique identifier for a request (e.g. can it be ignored)? Seems like it might be relevant if you were querying same model with same params but different store for some reason (but i'm not sure)? Maybe the store key needs to be treated specially in the stringifyObjectValues method, but that still leaves the problem of having other options that might have circular references breaking the code.

### PR DESCRIPTION
**Pull request by snewcomer**
@artinboghosian 👀 

close #360

### CODE REVIEW

File: addon/services/infinity.js [hunk 1]
```diff
@@ -35,21 +35,31 @@ let cacheInfinityCollection = (_cachedCollection, infinityModel, identifier, tim
...
  * @return String
  */
 let stringifyObjectValues = (options, identifier = '') => {
-  return Object.keys(options).reduce((acc, key) => {
-    const value = options[key];
-    if (!!value && typeof value === 'object') {
-      return stringifyObjectValues(value, acc);
-    }
+  return Object.keys(options)
+    .filter((key) => ALLOWED_KEYS.indexOf(key) > -1 || typeof options[key] === 'string')
```
Comment (artinboghosian) [inline addon/services/infinity.js]: What if value is number or boolean (e.g. `this.model.infinity('model-name', { foo: 1, bar: true })`)? We may want a list of acceptable types as well.

Comment (snewcomer) [reply addon/services/infinity.js]: good idea...fixed!

**Comment by snewcomer** [comment]
@artinboghosian v 1.3.3 published.  Thank you so much for the bug report and feedback!!

**Comment by artinboghosian** [comment]
@snewcomer np, thanks for quick turn around as well.

**Pull request merged**